prediction_market_agent/agents/microchain_agent/utils.py

```diff
@@ -54,9 +54,10 @@ def get_binary_markets(market_type: MarketType) -> list[AgentMarket]:
 def get_balance(market_type: MarketType) -> BetAmount:
     currency = market_type.market_class.currency
     if market_type == MarketType.OMEN:
-        # We focus solely on xDAI balance for now to avoid the agent having to wrap/unwrap xDAI.
+        balances = get_balances(APIKeys().bet_from_address)
+        total_balance = balances.xdai + balances.wxdai
         return BetAmount(
-            amount=get_balances(APIKeys().bet_from_address).xdai,
+            amount=total_balance,
             currency=currency,
         )
```

Comment:
I find this a bit dangerous - indeed xDAI and wxDAI have the same value, but let's say the user has 10 xDAI and 20 wxDAI - he sees this balance and now wants to bet 30 units on a given market. Do we auto-trigger a conversion between them to allow for such a trade?
Another option would be to have 2 balances for trades (less nice in my opinion).

Reply:
Good point, @coderabbitai please create an issue to change all functions where `auto_deposit` is in the argument, to instead of depositing everything from xDai, deposit only the remainder of `available_wxdai - needed_wxdai`, for example here prediction_market_agent_tooling/markets/omen/omen.py

Reply:
> Do we auto-trigger a conversion between them to allow for such a trade?

Yes, we do! See the `auto_deposit` arg https://github.com/gnosis/prediction-market-agent-tooling/blob/1e497fff9f2b53e4e3e1beb5dda08b4d49da881b/prediction_market_agent_tooling/markets/omen/omen.py#L139 (as @kongzii mentions).

So it's safe, but yes https://github.com/gnosis/prediction-market-agent/issues/206 would be a good optimization